dashboard/src/components/relocation/Toast.js

```diff
@@ -0,0 +1,223 @@
+      paddingLeft: 24,
+      paddingRight: 16,
+      marginRight: -8,
+
+      [theme.breakpoints.down("md")]: {
+        marginRight: "env(safe-area-inset-right)",
+      },
+    },
+
+    success: {
+      backgroundColor: green[600],
+    },
+    error: {
+      backgroundColor: theme.palette.error.dark,
+    },
+    info: {
+      backgroundColor: theme.palette.primary.dark,
+    },
+    warning: {
+      backgroundColor: amber[700],
+    },
+    icon: {
+      fontSize: 20,
+    },
+    variantIcon: {
+      opacity: 0.9,
+      fontSize: 20,
+      marginRight: theme.spacing.unit,
+    },
+  };
+};
+
+class Toast extends React.PureComponent {
+  static propTypes = {
+    classes: PropTypes.object.isRequired,
+    message: PropTypes.node,
+    variant: PropTypes.oneOf(Object.keys(icons)),
+    onClose: PropTypes.func.isRequired,
+    maxAge: PropTypes.number,
+    progress: PropTypes.node,
+  };
+
+  static defaultProps = {
+    maxAge: 0,
+    variant: undefined,
+    message: undefined,
+    progress: undefined,
+  };
+
+  state = { mouseOver: false };
+
+  id = `Toast-${uniqueId()}`;
+
+  _handleMouseOver = () => {
+    this.setState(state => {
+      if (state.mouseOver) {
+        return null;
+      }
+
+      return { mouseOver: true };
+    });
+  };
+
+  _handleMouseLeave = () => {
+    this.setState(state => {
+      if (!state.mouseOver) {
+        return null;
+      }
+
+      return { mouseOver: false };
+    });
+  };
+
+  render() {
+    const {
+      classes,
+      message,
+      onClose,
+      variant,
+      maxAge,
+      progress: progressBar,
+    } = this.props;
+
+    const { mouseOver } = this.state;
+
+    const Icon = icons[variant];
+
+    const messageId = `${this.id}-message`;
+
+    const closeButton = (
+      <IconButton
+        key="close"
+        aria-label="Close"
+        color="inherit"
+        className={classes.close}
+        onClick={onClose}
+      >
+        <CloseIcon className={classes.icon} />
+      </IconButton>
+    );
+
+    return (
+      <Paper
+        component={Typography}
+        headlineMapping={{
+          body1: "div",
+        }}
+        role="alertdialog"
+        square
+        elevation={6}
+        className={classNames(classes.root, classes[variant])}
+        aria-describedby={messageId}
+        onMouseOver={this._handleMouseOver}
+        onMouseLeave={this._handleMouseLeave}
+      >
+        <div className={classes.progress}>{progressBar}</div>
+        <div id={messageId} className={classes.message}>
+          {Icon && <Icon className={classes.variantIcon} />}
+          {message}
+        </div>
+        <div className={classes.action}>
+          {maxAge !== 0 ? (
+            <Timer
+              key={closeButton.props.key}
+              delay={maxAge}
+              onEnd={onClose}
+              paused={mouseOver}
+            >
+              {progress => (
+                <CircularProgress width={4} value={progress} opacity={0.5}>
```

Comment:
@jamesdphillips I thought it would help avoid user frustration when dealing with toasts disappearing before they can be interacted with, especially considering that the toast lifespan is variable.

Do you think it helps in this situation, or isn't worth the additional busyness?